### ISSUE TITLE
generate facets and annotations with mappings

### ISSUE THREAD

**Issue by amarant** [highsource/jsonix-schema-compiler#64]
Hi,
it would be great if the mappings could transport the xsd facets : minLength, pattern (even if they are not fully compatible with JS regex), ..., like it does with minLength/maxLength; and the annotations. I use the mappings to introspect a schema set, and it is far easier to use than the xsd mapped in json with jsonix.
It could be optional to not add weight to the mappings.
Thanks.

**Comment by highsource** [comment]
Which ones do you need, specifically? I think many are already exported, which ones are you missing?

**Comment by amarant** [comment]
Currently there is no facets on the data type I though :
https://www.w3.org/TR/xmlschema-2/#rf-facets

- length
- minLength
- maxLength
- pattern
- enumeration
- whiteSpace
- maxInclusive
- maxExclusive
- minExclusive
- minInclusive
- totalDigits
- fractionDigits

There is only element restriction, like minOccurs and maxOccurs :
https://www.w3.org/TR/2001/REC-xmlschema-0-20010502/#ref6

**Comment by gmoyanollc** [comment]
Is JAXB the reason why facets are missing? 

https://jaxb.java.net/guide/Generating_Schema_that_you_want.html 
https://java.net/jira/browse/JAXB-917

JAXB may be another casualty for the Java community as JavaScript has blossomed!

**Comment by mvlare** [comment]
Yes, i do see several similar other issue threads on missing facets minLength, maxLength, pattern. And i am also missing this!
